pygmt/datasets/map_tiles.py

```diff
@@ -0,0 +1,119 @@
+
+    source : xyzservices.TileProvider or str
+        [Optional. Default: Stamen Terrain web tiles] The tile source: web tile
+        provider or path to a local file. The web tile provider can be in the
+        form of a :class:`xyzservices.TileProvider` object or a URL. The
+        placeholders for the XYZ in the URL need to be {x}, {y}, {z},
+        respectively. For local file paths, the file is read with
+        :doc:`rasterio <rasterio:index>` and all bands are loaded into the
+        basemap. IMPORTANT: tiles are assumed to be in the Spherical Mercator
+        projection (EPSG:3857).
+
+    lonlat : bool
+        [Optional. Default: True]. If False, coordinates in ``region`` are
+        assumed to be Spherical Mercator as opposed to lon/lat.
+
+    wait : int
+        [Optional. Default: 0]. If the tile API is rate-limited, the number of
+        seconds to wait between a failed request and the next try.
+
+    max_retries : int
+        [Optional. Default: 2]. Total number of rejected requests allowed
+        before contextily will stop trying to fetch more tiles from a
+        rate-limited API.
+
+    kwargs : dict
+        Extra keyword arguments to pass to :func:`contextily.bounds2img`.
+
+    Returns
+    -------
+    raster : xarray.DataArray
+        Georefenced 3D data array of RGB value.
+
+    Raises
+    ------
+    ModuleNotFoundError
+        If ``contextily`` is not installed. Follow
+        :doc:`install instructions for contextily <contextily:index>`, (e.g.
+        via ``pip install contextily``) before using this function.
+
+    Examples
+    --------
+    >>> import contextily
+    >>> from pygmt.datasets import load_map_tiles
+    >>> raster = load_map_tiles(
+    ...     region=[103.60, 104.06, 1.22, 1.49],  # West, East, South, North
+    ...     source=contextily.providers.Stamen.TerrainBackground,
+    ...     lonlat=True,  # bounding box coordinates are longitude/latitude
+    ... )
+    >>> raster.sizes
+    Frozen({'band': 3, 'y': 1024, 'x': 1536})
+    >>> raster.coords
+    Coordinates:
+      * band     (band) int64 0 1 2
+      * y        (y) float64 1.663e+05 1.663e+05 1.663e+05 ... 1.272e+05 ...
+      * x        (x) float64 1.153e+07 1.153e+07 1.153e+07 ... 1.158e+07 ...
+    """
+    if contextily is None:
+        raise ModuleNotFoundError(
+            "Package `contextily` is required to be installed to use this function. "
+            "Please use `pip install contextily` or "
+            "`conda install -c conda-forge contextily` "
+            "to install the package"
+        )
+
+    west, east, south, north = region
+    image, (left, right, bottom, top) = contextily.bounds2img(
+        w=west, s=south, e=east, n=north, source=source, ll=lonlat, **kwargs
+    )
+
+    # Turn RGBA image from channel-last (H, W, C) to channel-first (C, H, W)
+    # and get just RGB (3 band) by dropping RGBA's alpha channel
```

Comment:
Why drop the alpha channel? GMT supports GeoTiff files with alpha channel.

Reply:
I think the alpha channel is the same throughout (at least for the basemaps I tried before), so not very helpful.

Reply:
Looking at the [grdimage_img_c2s_with_intensity](https://github.com/GenericMappingTools/gmt/blob/855167efa1d2b58debdc3a81e0823c1892acfa67/src/grdimage.c#L979) function, it can handle 4 bands with transparency.

Reply:
Ok, but the function returns an `xarray.DataArray`, not a GeoTIFF. The current `pygmt.grdimage` doesn't actually support plotting 4-band `xarray.DataArray`s (or even 3-band ones, see #1555) yet. So I don't see how this alpha channel can realistically be used unless people know how to save the `xarray.DataArray` to a 4-band GeoTIFF with the bands as a proper color interpretation first (requires https://github.com/corteva/rioxarray/pull/414 most likely).

That said, there are only a few XYZ tiles that I know of which have actual transparencies, e.g. the Stamen Toner Labels example at https://github.com/geopandas/contextily/pull/114#issuecomment-562870882. Regular tiles like Stamen Terrain do have an alpha channel, but they don't really carry meaningful transparency information (i.e. they are all 0s or 1s).